third_party/terraform/utils/transport.go

```diff
@@ -41,6 +41,10 @@ func sendRequestWithTimeout(config *Config, method, rawurl string, body map[stri
 	reqHeaders.Set("User-Agent", config.userAgent)
 	reqHeaders.Set("Content-Type", "application/json")
 
+	if timeout == 0 {
```

Comment:
Why not just have `sendRequest` send 5 instead of 0?

Reply:
I had considered that, but if somebody called sendRequestWithTimeout and passed in a 0 timeout (the nil value for timeout) then it will timeout immediately which doesn't feel like a good behavior for handling a nil value. I would have preferred if a nil value caused there to be no timeout at all but that would have made the method a bit more complex so I took this middle route. 

I don't feel super strongly about any of the options so let me know if you do.

Reply:
Drive-by but personally I would prefer passing in the default value in `sendRequest` instead of replacing actual timeout values of 0 with that default

Reply:
Out of curiosity, what's your opinion on expected behavior when passing in a nil value into the callWithTimeout method?

Reply:
Either an explicit error or something terrible happening! I could also see one of `0` or `nil` meaning no timeout at all if `time.Duration` doesn't express that in another way.

Reply:
Passing actual nil would be compile time, but it's recommended that you code should handle zero value inputs in a responsible manner. So I would think that if you passed in a zero value for a timeout that the code would handle it with no timeout, since a timeout of 0 isn't really a thing.
Kind of like how methods that are on pointer receivers shouldn't throw errors or panic when called with a nil receiver.

Reply:
That makes sense. I don't think there's a way to do infinity, but you can certainly just do a very large number.